osmomath/math.go

```diff
@@ -203,3 +206,62 @@ func OrderOfMagnitude(num Dec) int {
 
 	return order
 }
+
+var legacyPrecisionOf5sReuse = new(big.Int).Exp(big.NewInt(5), big.NewInt(sdkmath.LegacyPrecision), nil)
+var twoFactorsToRemovePerStep = uint(sdkmath.LegacyPrecision) - 1
+var divisorsNoPowersOfTwo = []*big.Int{nil}
+
+func init() {
+	for i := 0; i < 5; i++ {
+		cur := new(big.Int).Exp(legacyPrecisionOf5sReuse, big.NewInt(int64(i+1)), nil)
+		divisorsNoPowersOfTwo = append(divisorsNoPowersOfTwo, cur)
+	}
+}
+
+// LegacyMultiMulMut does the equivalent of base * exps[0] * exps[1] * ... * exps[n]
+// while mutating base.
+//
+// Right now this is done via enlarging the base, and doing a single "chop precision and round" for the batch.
+// Ideally in the future we do something smarter that intelligently figures out when to chop precision and round
+// based on the magnitudes/growth of the intermediate result.
+func LegacyMultiMulMut(base Dec, exps ...Dec) Dec {
```

Comment:
Am not sure that "exps" is an appropriate name for multiplicands, they aren't "exponents"; perhaps "mults"?